foam_controlwrapper/foam_files.py

```diff
@@ -80,14 +112,444 @@ def writeFoamFiles(self, caseDirectory, solver):
+        fileContent = self.create_file_content(caseDirectory, solver,
+                                               time, writeFields)
         for file in fileContent:
             try:
-                f = open(os.path.join(caseDirectory, file), 'w')
-                f.write(fileContent[file])
-                f.close()
+                full_name = os.path.join(caseDirectory, file)
+                with open(full_name, 'w') as f:
+                    f.write(fileContent[file])
             except IOError:
                 error_str = "Can't write file: {}"
-                raise ValueError(error_str.format(os.path.join(caseDirectory,
-                                                               file)))
+                raise IOError(error_str.format(os.path.join(caseDirectory,
+                                                            file)))
+
+    def modify_files(self, case, startTime, SP, BC, solver, SPExt, CMExt):
+        """Modify OpenFoam case files according to user settings
+
+        Parameters
+        ----------
+        case : str
+            case directory
+        startTime : str
+            start time
+        SP : DataContainer
+            System Parameters
+        BC : DataContainer
+            Boundary Conditions
+        solver : str
+            solver name
+        SPExt : dictionary
+            extension to SP
+
+        CMExt : dictionary
+            extension to CM
+
+        """
+
+        nOfTimeSteps = SP[CUBA.NUMBER_OF_TIME_STEPS]
+        deltaT = SP[CUBA.TIME_STEP]
+        endTime = int(startTime) + nOfTimeSteps*deltaT
+        writeInterval = endTime-int(startTime)
+        # if empty type boundary condition is used this must be
+        # changed in foam's polyMesh/boundary file to the same
+        # type (default type is patch)
+        pressureBCs = BC[CUBA.PRESSURE]
+        emptyBoundaries = []
+        for boundary in pressureBCs:
+            if pressureBCs[boundary] == "empty":
+                emptyBoundaries.append(boundary)
+
+        boundaryFile = os.path.join(case, 'constant', 'polyMesh', 'boundary')
+        try:
+            control = ParsedParameterFile(boundaryFile, boundaryDict=True)
+            boundaries = control.content
+            for boundary in emptyBoundaries:
+                for bi in range(len(boundaries)):
+                    filb = boundaries[bi]
+                    if filb == boundary:
+                        boundaries[bi+1]['type'] = "empty"
+                        break
+
+        except IOError:
+            error_str = "File {} does not exist"
+            raise IOError(error_str.format(boundaryFile))
+        try:
+            control.writeFile()
+        except IOError:
+            error_str = "Can't write file with content: {}"
+            raise IOError(error_str.format(control))
+
+# parse system/controlDict -file in case directory
+        parFile = os.path.join(case, 'system', 'controlDict')
+        try:
```

Comment:
I do not think that you need to use the try..except blocks here, the original error should contain all the necessary information.